models/tokens/solana/tokens_solana_transfers.sql

```diff
@@ -14,56 +14,235 @@
                                     \'["ilemi"]\') }}')
 }}
 
-SELECT
-    call_block_time as block_time
-    , cast (date_trunc('day', call_block_time) as date) as block_date
-    , call_block_slot as block_slot
-    , action
-    , amount
-    , COALESCE(tk_s.token_mint_address, tk_d.token_mint_address) as token_mint_address
-    , tk_s.token_balance_owner as from_owner
-    , tk_d.token_balance_owner as to_owner
-    , account_source as from_token_account
-    , account_destination as to_token_account
-    , call_tx_signer as tx_signer
-    , call_tx_id as tx_id
-    , call_outer_instruction_index as outer_instruction_index
-    , COALESCE(call_inner_instruction_index,0) as inner_instruction_index
-    , call_outer_executing_account as outer_executing_account
-FROM (  
-      SELECT account_source, account_destination, amount, call_tx_id, call_block_time, call_block_slot, call_outer_executing_account, call_tx_signer, 'transfer' as action, call_outer_instruction_index, call_inner_instruction_index
+with 
+token2022_fee_state as (
+      --we need the fee basis points and maximum fee for token2022 transfers because the fee amount is not emitted in transferChecked
+      SELECT 
+      call_account_arguments[1] as account_mint
+      , try(bytearray_to_uint256(bytearray_reverse(bytearray_substring(call_data,
+                  1+1+1+1+1+case when bytearray_substring(call_data,1+1+1,1) = 0x01 and bytearray_substring(call_data,1+1+1+32+1,1) = 0x01
```

Comment:
this looks painful 🙈

Reply:
could this be turned into a UDF?

Reply:
Haha maybe one day when we have a solana udf. Nothing that can be done for now